x-pack/legacy/plugins/siem/public/pages/case/components/tag_list/index.test.tsx

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { mount } from 'enzyme';
+
+import { TagList } from './';
+import { getFormMock } from '../__mock__/form';
+import { TestProviders } from '../../../../mock';
+import { wait } from '../../../../lib/helpers';
+import { useForm } from '../../../../../../../../../src/plugins/es_ui_shared/static/forms/hook_form_lib/hooks';
+import { act } from 'react-dom/test-utils';
+jest.mock(
+  '../../../../../../../../../src/plugins/es_ui_shared/static/forms/hook_form_lib/hooks/use_form'
+);
+const onSubmit = jest.fn();
+const defaultProps = {
+  disabled: false,
+  isLoading: false,
+  onSubmit,
+  tags: [],
+};
+
+describe('TagList ', () => {
```

Comment:
I think we can add a test that simulates the following scenario: The user press the edit button, adds a tag but then she presses cancel.

Reply:
Initial render test (without editing the tags) :)

Reply:
idk about all these initial rendering... if the sub components im interacting with exist, the entire component exists
```
    expect(
      wrapper
        .find(`[data-test-subj="no-tags"]`)
        .last()
        .exists()
    ).toBeTruthy();
```

Reply:
Hi! The initial rendering test checks how the component behaves before all data have been fetched (defaults passed down as props). A component could have three phases: 1) No data - default state 2) Happy path (all data fetched and presented) 3) Unhappy path (errors).

I agree that if the sub-components you are interacting exist then the entire component exists. It's just I cannot see this logic separated. Maybe the first test, in this file, could be split up in two. One that checks only 

> expect(
>       wrapper
>         .find(`[data-test-subj="no-tags"]`)
>         .last()
>         .exists()
>     ).toBeTruthy();

and one that does the clicking at checking the other components. 

Your checks test for existence in an indirect way so it's up to you :)

Reply:
i disagree, i want to avoid mounting more than i have to. if it fails on the render, we find out because the error is caught before the click. if it fails on the click, we know its the click. whats the problem?

Reply:
I think to make everyone happy here we can put the wrapper in a `beforeEach` and add specific test for rendering, edit, deleting etc... I think it is best to test one thing at the time.